cfme/infrastructure/host.py

```diff
@@ -485,11 +493,14 @@ def create(self, name, provider=None, credentials=None, hostname=None, ip_addres
         view.flash.assert_success_message(flash_message)
         return host
 
-    def all(self, provider):
+    def all(self):
```

Comment:
**REQUIRED:** So we are dropping the provider requirement and simply using the parent as the ```provider```, but the problem is that if we access the hosts collection from the appliance, then the parent will be an appliance object which will not work. I also think that given the simplicity of the function, we could perhaps also look to obtaining it via REST instead of the UI, which would be much quicker. We can also then think about dropping hosts from the appliance list of collections, and simple instantiating it as a provider based collection as you have done here. Obviously it is applicable to obtain a host without a provider, the problem is _as I remember_ that our Host object doesn't allow you to instantiate it without giving a provider.

Reply:
@psav 
* Testing this PR I was able successfully to get:
```python
hosts = appliance.collections.hosts.all()
```
as well as
```python
hosts = provider.hosts.all()
```
because in the first case `self.filters.get('parent') == None`
* `provider` is not required argument, so a host can be successfully instantiated without it
* if we drop the hosts collections the only way to get all aplliance's hosts will be iterating through all providers:
```python
hosts = [host for host in provider.hosts.all() for provider in appliance.collections.infra_providers.all()]
```
* Agree with REST, I will rework that.

Reply:
Ahhhhhh, for some reason i thought host required provider!!
Yeh lets get REST in, should be simple.

Reply:
@psav it seems there is no way to get host of a provider via REST:
```python
In [58]: appliance.rest_api.collections.providers[0].subcollections
Out[58]:
[u'cloud_networks',
 u'cloud_subnets',
 u'cloud_tenants',
 u'custom_attributes',
 u'flavors',
 u'load_balancers',
 u'policies',
 u'policy_profiles',
 u'security_groups',
 u'tags',
 u'vms']
```